### PR TITLE
[5.1] Allow PDF embeding again

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #43407 & #43365 .

### Summary of Changes
The security hardening adds an empty "sandbox" attribute to every iframe ([read more](https://web.dev/articles/sandboxed-iframes#granular_control_over_capabilities)). This prevents PDFs insert by the media manager loaded (it needs more fine granulated sandbox values)

With TinyMCE 7 you can [whitelist certain domains](https://www.tiny.cloud/docs/tinymce/latest/content-filtering/#sandbox-iframes-exclusions), which will overcome this problem, but till then, this filter is super strict and not usable.

### Testing Instructions
Use the media manager to embed a PDF. Save the article and open article in frontpage.

### Actual result BEFORE applying this Pull Request
- PDF is blocked

### Expected result AFTER applying this Pull Request
- Go to the TinyMCE configuration and disable the following option:
grafik

- PDF is visible

### CODE REVIEW

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on 46a404f4e460f49d9832844ea4c9ee8a9ef0179b<br /><br />Thank you<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

**Comment by SniperSister** [comment]
> Revert broken security fix 

It's not broken and it's not a "bug". It's exactly doing what it's supposed to do: it limits the possible actions that an iframe can perform to protect the user from unexpected script execution.

So, the question actually is: does the CMS want to enforce the strictest possible security configuration for the editor, at least by default? Or is the comfort in the content creation process more important?

That's a question that's probably worth a vote in the production department meeting.

**Comment by drmenzelit** [comment]
I'm for security, of course, but it should at least be possible to embed a local PDF ...
https://github.com/joomla/joomla-cms/issues/43407#issuecomment-2140121161

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 46a404f4e460f49d9832844ea4c9ee8a9ef0179b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

**Comment by Fedik** [comment]
TBH, I have no idea why it is called security,
Crossorigin is not easy to bypass without extra effort, by just embed iframe,
And all local embendings, well, they will be need to be enabled in one or in another way. 

We just making life for average User more complicated. They can just switch to codemirror, or none editor, and all this "theorethical security" will disapears.

**Comment by SniperSister** [comment]
We are not talking about a CORS but about user experience and perception. Imagine an editor copy&pasting an iframe tag to an external site into the editor window of his companies intranet. On paste, the iframe will be loaded automatically - and without sandboxing, that iframe could i.e. trigger a file download. The user is now under the impression that this download is triggered by a trusted source, which is the Joomla administrator interface of his intranet - whereas the download is actually triggered by the untrusted iframe site.

I'm not necessarily saying, that this is an important threat scenario for the average user, I just want to point out that the issue is not related to crossorigin or samesite policies.

**Comment by Fedik** [comment]
We can make a TinyMCE plugin, that shows an alert for potentialy dangerous iframe, when User insert something in the editor. instead of totaly blocking everything.

**Comment by RickR2H** [comment]
Maybe its an option to extend the TinyMCE plugin with the basic configuration for the sandbox parameters so a domain could be added to allow the PDF iframe?

**Comment by bembelimen** [comment]
Now with parameter. Ready to test again :)

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 12ac335d94127a898fd7e4cebce70cb1e5839a07<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

**Comment by SniperSister** [comment]
I have tested this item :white_check_mark: successfully on 12ac335d94127a898fd7e4cebce70cb1e5839a07<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43716">issues.joomla.org/tracker/joomla-cms/43716</a>.</sub>

File: administrator/language/en-GB/plg_editors_tinymce.ini [hunk 1]
```diff
@@ -96,4 +96,6 @@ PLG_TINY_TOOLBAR_BUTTON_FONTSIZESELECT="Font Size Select"
 PLG_TINY_TOOLBAR_BUTTON_FORMATSELECT="Format Select"
 PLG_TINY_TOOLBAR_BUTTON_SEPARATOR="Separator"
 PLG_TINY_TOOLBAR_BUTTON_STYLESELECT="Style Select"
+PLG_TINY_FIELD_SANDBOX_IFRAMES_LABEL="Sandbox Iframes"
+PLG_TINY_FIELD_SANDBOX_IFRAMES_DESC="This is a security feature that restricts what the iframe can do by setting a sandbox attribute for each iframe. It is recommended to enable this feature for security reasons. Disable only if the iframe is not loading and you're sure that the content iframed is safe."
```
Comment (brianteeman) [inline administrator/language/en-GB/plg_editors_tinymce.ini]:
```suggestion
PLG_TINY_FIELD_SANDBOX_IFRAMES_DESC="This is a security feature that restricts what the iframe can do by setting a sandbox attribute for each iframe. It is recommended to enable this feature for security reasons. Only disable if the iframe is not loading and you're sure that the iframed content is safe."
```

**Comment by LadySolveig** [comment]
Thank you @bembelimen and also for testing and review @drmenzelit @SniperSister @RickR2H @Fedik @brianteeman

**Comment by niosme** [comment]
Does this work with iframes like google maps iframe inside editor?

**Comment by DeMiro5001** [comment]
I am not sure to understand why using iframe tag instead of embed tag for pdf.

**Pull request merged**